test/demo/open.test.ts

```diff
@@ -0,0 +1,171 @@
+                // Check that both apps are running
+                await expect(fin.Application.wrapSync({uuid: 'test-app-1'}).isRunning()).resolves.toBe(true);
+                await expect(fin.Application.wrapSync({uuid: 'test-app-2'}).isRunning()).resolves.toBe(true);
+            }, 10000);
+        });
+    });
+
+    describe('With context', () => {
+        const validContext: OrganizationContext = {type: 'organization', name: 'OpenFin', id: {default: 'openfin'}};
+        const testAppIdentity1: Identity = {uuid: 'test-app-preregistered-1', name: 'test-app-preregistered-1'};
+        const testAppIdentity2: Identity = {uuid: 'test-app-preregistered-2', name: 'test-app-preregistered-2'};
+
+        afterEach(async () => {
+            // Close all test apps and suppress any errors since the apps may not be running
+            await fin.Application.wrapSync(testAppIdentity1).quit().catch(() => {});
+            await fin.Application.wrapSync(testAppIdentity2).quit().catch(() => {});
+        });
+
+        test('When passing a valid app name and a valid context, the app opens and its context listener is triggered with the correct data', async () => {
+            // From the launcher app, call fdc3.open with a valid name and context
+            await fdc3Remote.open(testManagerIdentity, 'test-app-preregistered-1', validContext);
+
+            // Check that the app is now running
+            await expect(fin.Application.wrapSync({uuid: 'test-app-preregistered-1'}).isRunning()).resolves.toBe(true);
+
+            // Retrieve the list of contexts the app received
+            const preregisteredListener = await fdc3Remote.getRemoteContextListener(testAppIdentity1);
+            const receivedContexts: Context[] = await preregisteredListener.getReceivedContexts();
+
+            // Check that the app received the context passed in open and nothing else
+            expect(receivedContexts.length).toBe(1);
+            expect(receivedContexts.slice(-1)[0]).toEqual(validContext);
+        });
+
+        test.todo('When passing a known app name but invalid context, [behaviour TBD]');
+
+        test('When passing an unknown app name with any context the service returns an error', async () => {
+            // From the launcher app, call fdc3.open with an invalid name and valid context
+            // TODO: fill in the error message once proider work is done and we know what it will be (SERVICE-392)
+            await expect(fdc3Remote.open(testManagerIdentity, 'invalid-app-name', validContext)).rejects.toThrowError();
+        });
+
+        describe('With an app already running', () => {
+            beforeEach(async () => {
+                // Start by opening a test app
+                await fdc3Remote.open(testManagerIdentity, 'test-app-preregistered-1');
+            });
+
+            test(
+                'When opening the running app it is focused, its context listener is triggered with the correct data, and the promise resolves',
```

Comment:
So for the highlighted case it would be:
`
describe('Opening apps....', () => {
  describe('With context', () => {
    describe('App already running', () => {
      beforeEach() // Open app1
      describe('Opening the already running app', () => {
        beforeEach() // Open app1
        test('The app is focused and its listener triggered')
      })
      describe('Opening a second app', () => {
        beforeEach() // Open app2
        test('Second app opens and listener is triggered')
      })
})
`